assets/js/components/Portal.js

```diff
@@ -20,18 +20,28 @@
  * External dependencies
  */
 import { useEffectOnce } from 'react-use';
+import PropTypes from 'prop-types';
 
 /**
  * WordPress dependencies
  */
 import { createPortal, useState } from '@wordpress/element';
 
-function Modal( { children } ) {
+function Portal( { children, slug, appendPluginRoot } ) {
+	const elDiv = document.createElement( 'div' );
+	if ( slug ) {
+		elDiv.classList.add( `googlesitekit-portal-${ slug }` );
+	}
+
+	if ( ! appendPluginRoot ) {
+		elDiv.classList.add( 'googlesitekit-plugin' );
+	}
+
 	// Using state as we need `el` to not change when the component re-renders
-	const [ el ] = useState( document.createElement( 'div' ) );
+	const [ el ] = useState( elDiv );
 
 	useEffectOnce( () => {
-		const root = document.querySelector( '.googlesitekit-plugin' ) || document.body;
+		const root = appendPluginRoot ? document.querySelector( '.googlesitekit-plugin' ) || document.body : document.body;
```

Comment:
I think we don't need it and can continue adding "portals" to the `.googlesitekit-plugin` element as we do it right now. I know that it was requested in the AC, but the main intention of it is to make sure that we display the survey popup in the bottom-right corner and ignore parent containers. It can be easily achieved by using `position: fixed` as you do it for the survey popup and that's all that we need. So, let's revert `appendPluginRoot` modifications.

Reply:
@eugene-manuilov I think we should keep it since CSS works in that case but I do think later down the road where we'd want that functionality. And I think having the flexibility is a plus compared to not having it.

Reply:
> I do think later down the road where we'd want that functionality

It is "premature optimization" which is usually called "the root of all evil". We don't have a real use case for it for now and there is nothing on the horizon that might need it. Thus, let's not add it.

> And I think having the flexibility is a plus compared to not having it.

Yes, we are still flexible: if we need it in the future, we will update this component to let us select where to mount a portal.